### PR TITLE
Feature/907 accordion component

### PR DESCRIPTION
**Pull request by anna-lach**

### CODE REVIEW

File: packages/components/accordion/src/accordion-item.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+   */
+  #accordionItemId = `sl-accordion-item-${nextUniqueId++}`;
+
+  /**
+   * Unique ID for each accordion item component content.
+   */
+  #accordionItemContentId = `sl-accordion-item-content-${nextUniqueId++}`;
+
+  /** A text shown in the header - as a title of the accordion item. */
+  @property() summary!: string;
```
Comment (jpzwarte) [inline packages/components/accordion/src/accordion-item.ts:32]: Why the exclamation mark here? In other places we just do `summary?: string`

File: packages/components/accordion/src/accordion-item.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+  /** Whether the element is disabled. */
+  @property({ type: Boolean, reflect: true }) disabled?: boolean;
+
+  /** Emits when the accordion item has been toggled. */
+  @event({ name: 'sl-toggle' }) toggleEvent!: EventEmitter<string>;
+
+  /** Whether the details element is opened. */
+  @property({ reflect: true, type: Boolean }) open?: boolean;
```
Comment (jpzwarte) [inline packages/components/accordion/src/accordion-item.ts:41]: Nitpicking, but I try to keep the variables alphabetically sorted. So `toggleEvent` after `open` instead of before

File: packages/components/accordion/src/accordion-item.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+ */
+@localized()
+export class AccordionItem extends LitElement {
+  /** @private */
+  static override styles: CSSResultGroup = styles;
+
+  /**
+   * Unique ID for each accordion item component.
+   */
+  #accordionItemId = `sl-accordion-item-${nextUniqueId++}`;
```
Comment (jpzwarte) [inline packages/components/accordion/src/accordion-item.ts:24]: why the `accordionItem` prefix?

File: packages/components/accordion/src/accordion-item.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+  /**
+   * Unique ID for each accordion item component content.
+   */
+  #accordionItemContentId = `sl-accordion-item-content-${nextUniqueId++}`;
```
Comment (jpzwarte) [inline packages/components/accordion/src/accordion-item.ts:29]: Same

File: packages/components/accordion/src/accordion-item.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  @property({ type: Boolean, reflect: true }) disabled?: boolean;
+
+  /** Emits when the accordion item has been toggled. */
+  @event({ name: 'sl-toggle' }) toggleEvent!: EventEmitter<string>;
+
+  /** Whether the details element is opened. */
+  @property({ reflect: true, type: Boolean }) open?: boolean;
+
+  /** @private */
+  @query('.wrapper') wrapper!: HTMLDivElement;
```
Comment (jpzwarte) [inline packages/components/accordion/src/accordion-item.ts:44]: Since we only use this variable in the `#onClick` method, it might be better to just do `const wrapper = this.renderRoot.querySelector('.wrapper')` at the top of the `#onClick` method?

File: packages/components/accordion/src/accordion-item.ts [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+  override render(): TemplateResult {
+    return html`
+      <details @toggle=${this.#onToggle} ?open=${this.open}>
+        <summary
+          id=${this.#accordionItemId}
```
Comment (jpzwarte) [inline packages/components/accordion/src/accordion-item.ts:50]: Do we actually need unique ids *inside the shadow DOM*? iirc in other places, I don't do that. In `dialog.ts` for example: `<slot name="title" id="title"></slot>`

File: packages/components/accordion/src/accordion.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      this.items.forEach(item => {
+        if (item.renderRoot.querySelector('details')?.hasAttribute('open')) {
+          item.renderRoot.querySelector('summary')?.click();
+        }
+      });
+    }
+  }
+
+  override render(): TemplateResult {
+    return html` <slot @click=${this.#onClick}></slot> `;
```
Comment (jpzwarte) [inline packages/components/accordion/src/accordion.ts:46]: Unnecessary whitespace

File: packages/components/accordion/src/accordion.ts [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  #onClick(event: Event): void {
+    if (!this.single || event.defaultPrevented) {
+      // No toggling when no `single` or the user prevents it.
+      return;
+    }
+
+    this.items.forEach(item => {
+      if (item !== event.target) {
```
Comment (jpzwarte) [inline packages/components/accordion/src/accordion.ts:56]: This could probably be a `.filter(item => item !== event.target)` instead of an if statement?

**Review by jpzwarte — APPROVED** [review]

File: packages/components/accordion/src/accordion.ts [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+ *   <sl-accordionn>...</sl-accordion>
+ * ```
+ *
+ * @slot default - The place for multiple <sl-accordion-item>
+ */
+export class Accordion extends ScopedElementsMixin(LitElement) {
+  /** @private */
+  static get scopedElements(): ScopedElementsMap {
+    return {
+      'sl-accordion-item': AccordionItem
```
[the remark this answers is not in the document]
Comment (anna-lach) [reply packages/components/accordion/src/accordion.ts:20]: right :)

File: packages/components/accordion/src/accordion-item.ts (hunk 1 above)
Comment (anna-lach) [reply packages/components/accordion/src/accordion-item.ts:32]: Because summary should be always added, so I thought that we can assume that it's always there.

Comment (jpzwarte) [reply packages/components/accordion/src/accordion-item.ts:32]: We can't really. We can't force the user to set the `summary` property. In some places you can assume that something is set, but in this case it's dangerous.

File: packages/components/accordion/src/accordion-item.spec.ts [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+      await el.updateComplete;
+      await new Promise(resolve => setTimeout(resolve));
+
+      expect(details).to.have.attribute('open');
+      expect(summary).to.have.attribute('aria-expanded', 'true');
+    });
+
+    it('should open on Space', async () => {
+      summary.focus();
+      await sendKeys({ press: 'Enter' });
```
Comment (Diaan) [inline packages/components/accordion/src/accordion-item.spec.ts:65]:
```suggestion
      await sendKeys({ press: 'Space' });
```

File: packages/components/accordion/src/accordion-item.spec.ts [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+      await el.updateComplete;
+      await new Promise(resolve => setTimeout(resolve));
+
+      expect(details).not.to.have.attribute('open');
+      expect(summary).not.to.have.attribute('aria-expanded', 'true');
+    });
+
+    it('should ignore Space', async () => {
+      summary.focus();
+      await sendKeys({ press: 'Enter' });
```
Comment (Diaan) [inline packages/components/accordion/src/accordion-item.spec.ts:143]:
```suggestion
      await sendKeys({ press: 'Space' });
```

File: packages/components/accordion/src/accordion-item.ts (hunk 3 above)
Comment (anna-lach) [reply packages/components/accordion/src/accordion-item.ts]: removed :)

File: packages/components/accordion/src/accordion.ts [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+import { type CSSResultGroup, LitElement, type PropertyValues, type TemplateResult, html } from 'lit';
+import { property, queryAssignedElements } from 'lit/decorators.js';
+import { AccordionItem } from './accordion-item.js';
+import styles from './accordion.scss.js';
+
+/**
+ * An accordion component that can contain accordion-items
+ *
+ * ```html
+ *   <sl-accordionn>...</sl-accordion>
```
Comment (Diaan) [inline packages/components/accordion/src/accordion.ts:10]:
```suggestion
 *   <sl-accordion>...</sl-accordion>
```

File: website/package.json [hunk 13]
```diff
@@ -158,7 +158,7 @@
     "@11ty/eleventy-img": "^3.1.8",
     "@11ty/eleventy-navigation": "^0.3.5",
     "@11ty/eleventy-plugin-syntaxhighlight": "^5.0.0",
-    "@custom-elements-manifest/analyzer": "^0.9.3",
+    "@custom-elements-manifest/analyzer": "^0.9.2",
```
Comment (Diaan) [inline website/package.json:161]: is this deliberate or a merging mistake?

Comment (anna-lach) [reply website/package.json:161]: Good one! Merging mistake.

File: .changeset/six-mails-learn.md [hunk 14]
```diff
@@ -0,0 +1,5 @@
+---
+"@sl-design-system/accordion": minor
```
Comment (jpzwarte) [inline .changeset/six-mails-learn.md:2]: Please change this to `patch`
